components/optimizer/base.py

```diff
@@ -0,0 +1,40 @@
+import abc
+from common.minimodel import Group, Plan
+from components.selector import GroupInfo
+from typing import Dict
+
+Selection = Dict[Group, GroupInfo]
+
+
+class BaseOptimizer(metaclass=abc.ABCMeta):
```

Comment:
I'm not sure why you're using a `metaclass` here instead of inherting from `ABC`:

https://stackoverflow.com/questions/33335005/is-there-any-difference-between-using-abc-vs-abcmeta